Patch developed by Kurt Lust for LUMI.
Basically, the makefiles are junk as they fix the installation directory
and fix compiler names in a stupid way.
We also set CXX for QHull though for this code it is never used as only the
C-version of the library is built. But just for completeness, if this would
change in the future...
diff -Nau fpocket-4.2.2/makefile.orig fpocket-4.2.2/makefile
diff -Nau fpocket-4.2.2/src/qhull/Makefile.orig fpocket-4.2.2/src/qhull/Makefile

--- fpocket-4.2.2/makefile.orig	2024-09-09 15:20:23.000000000 +0300
+++ fpocket-4.2.2/makefile	2025-04-08 10:33:16.000000000 +0300
@@ -15,11 +15,10 @@
 PATH_QHULL  = src/qhull/src
 PATH_LIB	= $(PLUGINDIR)/$(ARCH)/molfile/
 
-BINDIR  = /usr/local/bin/
-LIBDIR  = /usr/local/lib/
-
-MANDIR  = /usr/local/man/man8/
-
+PREFIX  ?= /usr/local
+BINDIR   = $(PREFIX)/bin/
+LIBDIR   = $(PREFIX)/lib/
+MANDIR   = $(PREFIX)/share/man/man8/
 
 FPOCKET     = fpocket
 TPOCKET	    = tpocket
@@ -30,17 +29,11 @@
 PROGALL		= $(PATH_BIN)$(FPOCKET) $(PATH_BIN)$(TPOCKET) $(PATH_BIN)$(DPOCKET) $(PATH_BIN)$(MDPOCKET)
 PROGMDPOCKET    = $(PATH_BIN)$(MDPOCKET)
 
-ifeq ($(CXX),g++)
-  CC          = gcc
-  CCQHULL	    = gcc
-  LINKER      = LD_LIBRARY_PATH=$(PLUGINDIR)/$(ARCH)/molfile gcc
-  LINKERQHULL = gcc
-else 
-  CCQHULL	    = clang
-  CC          = clang
-  LINKER      = LD_LIBRARY_PATH=$(PLUGINDIR)/$(ARCH)/molfile clang
-  LINKERQHULL = clang
-endif
+CC           ?= gcc-14
+CXX          ?= g++-14
+CCQHULL       = $(CC)
+LINKER        = LD_LIBRARY_PATH=$(PLUGINDIR)/$(ARCH)/molfile $(CC)
+LINKERQHULL   = $(CC)
 
 CGSL        = -DMD_NOT_USE_GSL -I$(PATH_GSL)include
 COS         = -DM_OS_LINUX
@@ -48,7 +41,7 @@
 CWARN       = -W -Wextra -Wwrite-strings -Wstrict-prototypes
 
 CFLAGS      = $(CWARN) $(COS) $(CDEBUG) -O2 -g -pg -std=gnu99 -I$(PLUGINDIR)/include -I$(PLUGINDIR)/$(ARCH)/molfile
-QCFLAGS     = -O -g -pg -ansi
+QCFLAGS     = -O2 -g -pg -ansi
 
 LGSL        = -L$(PATH_GSL)lib -lgsl -lgslcblas 
 LFLAGS	    = -lm -L$(PLUGINDIR)/$(ARCH)/molfile $(PLUGINDIR)/$(ARCH)/molfile/libmolfile_plugin.a -lstdc++
@@ -160,7 +153,7 @@
 
 
 qhull:
-	cd src/qhull/ && make
+	cd src/qhull/ && make CC=$(CC) CXX=$(CXX) 
 
 $(PATH_BIN)$(CHECK): $(CHOBJ) $(QOBJS)
 	$(LINKER) $^ -o $@ $(LFLAGS)
@@ -179,6 +172,7 @@
 
 install:
 	mkdir -p $(BINDIR)
+	mkdir -p $(LIBDIR)
 	mkdir -p $(MANDIR)
 	cp $(PATH_BIN)$(FPOCKET) $(BINDIR)
 	cp $(PATH_BIN)$(TPOCKET) $(BINDIR)
--- fpocket-4.2.2/src/qhull/Makefile.orig	2024-09-09 15:20:23.000000000 +0300
+++ fpocket-4.2.2/src/qhull/Makefile	2025-04-08 10:33:29.000000000 +0300
@@ -90,16 +90,8 @@
 # PRINTMAN = lpr
 # PRINTC = lpr
 
-#for Gnu's gcc compiler, -O3 for optimization, -g for debugging, -pg for profiling
-# -fpic needed for gcc x86_64-linux-gnu.  Not needed for mingw
-ifeq ($(CXX),g++)
-  CC          = gcc
-  CXX		  = g++:
-else 
-  CC          = clang
-  CXX       = clang
-endif
-
+CC  ?= gcc-14
+CXX ?= g++-14
 
 CC_OPTS1  = -O3 -ansi -Isrc -fpic $(CC_WARNINGS)
 
